graphql/environment-nodes-query-1.graphql

```diff
@@ -0,0 +1,15 @@
+query {
+    environmentNodes(filter: { nodeName: "JDP" }) {
+        descendantTargets {
+            startRecording(recording: {
+                name: "foo",
```

Comment:
Ah, that's definitely somewhere to add better error handling. I think you are missing the JMX credentials required on some of the smoketest targets. The GraphQL API request generally completed and it was able to create some of the recordings you asked for, but there were some errors along the way for some targets. When you try to re-issue the request then all of the attempts fail either due to lacking credentials or if you do have credentials then because there is already a recording with the same name.

Just like for automated rules, currently you need to manually add credentials for such targets in advance using https://github.com/cryostatio/cryostat/blob/main/HTTP_API.md#TargetCredentialsPostHandler .

Reply:
You could test that that's the problem by changing the `environmentNodes` filter - instead of selecting the JDP node, try adding a custom target pointing either to Cryostat itself or to the `vertx-fib-demo:9093` target which doesn't require credentials (or both), and then select the node named `Custom Targets` instead of `JDP`.